distributed/tests/test_worker.py

```diff
@@ -2780,64 +2781,52 @@ async def test_forget_dependents_after_release(c, s, a):
     assert fut2.key not in {d.key for d in a.state.tasks[fut.key].dependents}
 
 
+@pytest.mark.filterwarnings("ignore:Large object of size")
 @gen_cluster(client=True)
 async def test_steal_during_task_deserialization(c, s, a, b, monkeypatch):
     stealing_ext = s.extensions["stealing"]
     await stealing_ext.stop()
-    from distributed.utils import ThreadPoolExecutor
 
-    class CountingThreadPool(ThreadPoolExecutor):
-        counter = 0
-
-        def submit(self, *args, **kwargs):
-            CountingThreadPool.counter += 1
-            return super().submit(*args, **kwargs)
-
-    # Ensure we're always offloading
-    monkeypatch.setattr("distributed.worker.OFFLOAD_THRESHOLD", 1)
-    threadpool = CountingThreadPool(
-        max_workers=1, thread_name_prefix="Counting-Offload-Threadpool"
-    )
-    try:
-        monkeypatch.setattr("distributed.utils._offload_executor", threadpool)
+    class SlowDeserializeCallable:
+        def __init__(self):
+            self.data = b"0" * (OFFLOAD_THRESHOLD + 1)
 
-        class SlowDeserializeCallable:
-            def __init__(self, delay=0.1):
-                self.delay = delay
+        def __getstate__(self):
+            return self.data
 
-            def __getstate__(self):
-                return self.delay
+        def __setstate__(self, state):
+            return SlowDeserializeCallable()
 
-            def __setstate__(self, state):
-                delay = state
-                import time
+        def __sizeof__(self):
+            return OFFLOAD_THRESHOLD * 2
 
-                time.sleep(delay)
-                return SlowDeserializeCallable(delay)
+        def __call__(self, *args, **kwargs):
+            return 41
 
-            def __call__(self, *args, **kwargs):
-                return 41
+    in_deserialize = asyncio.Event()
+    wait_in_deserialize = asyncio.Event()
 
-        slow_deserialized_func = SlowDeserializeCallable()
-        fut = c.submit(
-            slow_deserialized_func, 1, workers=[a.address], allow_other_workers=True
-        )
+    async def custom_worker_offload(func, *args):
+        res = func(*args)
+        if not istask(args) and istask(res):
+            in_deserialize.set()
+            await wait_in_deserialize.wait()
+        return res
 
-        while CountingThreadPool.counter == 0:
-            await asyncio.sleep(0)
+    monkeypatch.setattr("distributed.worker.offload", custom_worker_offload)
+    obj = SlowDeserializeCallable()
+    fut = c.submit(lambda _: 41, obj, workers=[a.address], allow_other_workers=True)
```

Comment:
What's the purpose of the custom class?
Wouldn't we get the same effect with `obj = b"0" * (OFFLOAD_THRESHOLD + 1)`?
Particularly its `__call__` method seems completely redundant.

Reply:
yes, this is a relict of the last test iteration. I'll simplify